src/index.tsx

```diff
@@ -1,58 +1,58 @@
 import * as React from 'react';
-import { getBase64, checkResolution } from './utils';
+import { triggerOpenDialog, getListFile, getAcceptString } from './utils';
+import {
+  isResolutionValid,
+  isImageValid,
+  maxFileSizeValid,
+  acceptTypeValid,
+  maxNumberValid,
+} from './validate';
 import {
   ImageType,
   ImageListType,
   ImageUploadingPropsType,
   ErrorsType,
   ResolutionType,
 } from './typings';
+import {
+  DEFAULT_NULL_INDEX,
+  INIT_ERROR,
+  INIT_MAX_NUMBER,
+  DEFAULT_DATA_URL_KEY,
+} from './constants';
 
-const { useRef, useState, useCallback } = React;
-
-const defaultErrors: ErrorsType = {
-  maxFileSize: false,
-  maxNumber: false,
-  acceptType: false,
-  resolution: false,
-};
-const defaultNullIndex = -1;
+const { useRef, useState, useCallback, useMemo } = React;
```

Comment:
We can use something like this now
```suggestion
import { useRef, useState, useCallback, useMemo }, React from 'react';
```